palm/plugins/dbt/commands/cmd_model-doc.py

```diff
@@ -0,0 +1,215 @@
+    target_file = model_name + ".yml"
+    destination = model_path.parent / target_file
+    if destination.exists():
+        click.echo(click.style(f"{model_name}.yml already exists", fg="yellow"))
+        overwrite = click.confirm("Do you want to overwrite it?", default=False)
+        if not overwrite:
+            return
+
+    model = {
+        "version": 2,
+        "models": [
+            {
+                "name": model_name,
+                "description": f'{{{{ doc("{model_name}") }}}}',
+                "columns": get_model_columns(model_path),
+            }
+        ],
+    }
+    destination.write_text(yaml.dump(model, sort_keys=False, default_flow_style=False))
+    click.echo(click.style(f"Model.yml generated at {destination}", fg="green"))
+    return destination
+
+
+def get_model_columns(model_path: Path) -> List[str]:
+    """Parse the model SQL file and return a list of columns names
+
+    Args:
+        model_path (Path): Path to the model SQL file
+
+    Returns:
+        List[str]: List of column names
+    """
+    raw = Path(model_path).read_text()
+    raw = re.sub(
+        r"{{[A-Za-z\\n\s()=_,]+}}", "", raw
+    ).strip()  # Strip out jinja config blocks
+    parsed = sqlparse.parse(raw)
+
+    column_identifiers = []
+
+    for statement in parsed:
+        # Unknown is acceptable here because dbt models aren't always proper SQL
+        if statement.get_type() in ["SELECT", "UNKNOWN"]:
+            column_identifiers = get_column_identifiers(statement)
+
+    column_names = get_column_names(column_identifiers)
+    columns = create_column_list(column_names)
+    return columns
+
+
+def get_column_identifiers(
+    statement: sqlparse.sql.Statement,
+) -> List[sqlparse.sql.Identifier]:
+    """Get the column identifiers from a SQL statement"""
+    identifiers = []
+    for token in statement.tokens:
+        if token.ttype is None and type(token) is not sqlparse.sql.Function:
+            if type(token) is sqlparse.sql.Identifier:
+                # If the token is a CTE, skip it
+                if not token_is_cte(token):
+                    identifiers.append(token)
+            elif type(token) is sqlparse.sql.IdentifierList:
+                for child_token in token.get_identifiers():
+                    if (
+                        child_token.ttype is not sqlparse.tokens.Keyword
+                        and not token_is_cte(child_token)
+                    ):
+                        identifiers.append(child_token)
+
+    if len(identifiers) == 0:
+        raise Exception("Could not find column identifiers")
+    return identifiers
+
+
+def token_is_cte(token: sqlparse.sql.Token) -> bool:
+    """Check if a given token is a CTE name so we can skip documenting it.
+
+    Args:
+        token (sqlparse.sql.Token): The token to check
+
+    Returns:
+        bool: True if the token is a CTE name, False otherwise
+    """
+    _, foo = token.token_next(1)
```

Comment:
was foo a test or something you wanted to leave in?

Reply:
No, good catch. This was previously called`next`, which caused problems in debugging. I need to give this a real name. Probably `next_token`